Topics/Tech_Stacks/Django_Rest.md

```diff
@@ -0,0 +1,120 @@
+# Building RESTful APIs using Django Rest Framework
+
+### Prerequisites
+
+Before learning Django Rest Framework, make sure that you have a working knowledge of the following:
+
+ - [Python](https://www.python.org/): High level general purpose programming language.
+ - [Django](https://www.djangoproject.com/): Web framework for Python.
+
+### What Is Django Rest Framework?
+Django Rest Framework is a popular library for building RESTful APIs with Django. It provides powerful tools for building APIs, including serialization, authentication, and permissions. You can easily create APIs that support multiple formats like JSON and XML that can be consumed by a variety of client applications. To demonstrate the power of Django Rest Framework, we will create a simple API.
+
+### Set-Up
+
+ We first set up the virtual environment and activate it using ```python3 -m venv myenv ``` and ```source myenv/bin/activate``` respectively in the project directory.
```

Comment:
Please either add a little bit more info about the constraints on setting up virtual env. When I was studying setting up virtual env with python, I had a problem where my python file isn't locating at the same directory as others and some of my peers had the same issue, but most tutorial doesn't mention this. It is critical to check out the directory for where the python file is located and make sure your virtual env points to the right location and the right version too. Please check out this resource for reference: https://help.dreamhost.com/hc/en-us/articles/115000695551-Installing-and-using-virtualenv-with-Python-3    (mainly from step 1 to 4)

Reply:
I also ran into this issue before. I added more details on setting up the virtual environment.